### ISSUE TITLE
Native Ipv6 support

### ISSUE THREAD

**Issue by troyBORG** [Yellow-Dog-Man/Resonite-Issues#143]
### Is your feature request related to a problem? Please describe.

This was a mod made by Rucio for the JP and KR users that don't have ipv4 in there country.   Or for people with ISP that have Carrier Grade NAT.

### Describe the solution you'd like

Allow users who have ipv6 to be able to connect to hosts who have ipv6 instead of only being ipv4 or having to use the LNL relay to connect.

### Describe alternatives you've considered

https://github.com/bontebok/ResoniteIPv6Mod

### Additional Context

This really need to be added into the game ASAP.

**Comment by Nammi-namm** [comment]
I am a fan of the idea of a simpler stopgap method (if this is even possible) to use the LNL relay servers as a form of connecting IPv4-only hosts with IPv6-only users. Or an IPv6-only host having all other users connect via a relay. 

This would be highly desirable since at least it means that if your connection is IPv4-only or IPv6-only you'll at least be able to connect to any session or have anyone connect to your session.

Even if IPv6 to IPv6 won't be natively supported, at least they'll be able to connect via a relay.

IPv6-only ISP's are a thing especially in Japan, where a big sizeable portion of the player base is from.

**Comment by troyBORG** [comment]
> But if native IPv6 functionality is easy to implement of course that should also be on the table.
Yeah it seems to work well with the mod that was made.  As long as the host and the user both had it installed.

I'm just wanting it to be native in the game so everyone can use it, instead of just hoping the person has the mod setup so the can connect.  Like my ISP gives me both a ipv4 and ipv6 address.  I have my headless setup with it, and I know the Creator Jam Headless also runs it, as the person who made the mod is the person who runs that headless.

**Comment by bontebok** [comment]
@troyBORG Hi Troy, almost missed this until I saw it go onto @ProbablePrime's board. To be clear, I did not create the [Resonite IPv6 Mod](https://github.com/bontebok/ResoniteIPv6Mod) for a specific community but for anyone that wanted native IPv6 connectivity in Resonite. This was driven mostly by ISPs that provide IPv4 using CGNat (Carrier Grade NAT). This is quite common on cellular providers, and in countries where IPv4 is more limited. With CGNat, you cannot connect to other Resonite sessions without going through a relay.

As Resonite has added more regional relays and the ability to adjust the LNL Window, less users behind CGNat situations are using the IPv6 Mod and relying more on the Resonite relays. However, native IPv6 connectivity would still be welcomed as there will be a time where IPv4 will begin to become secondary or come at a premium cost. (Pretty much all cloud providers are now charging for IPv4 as a premium service but giving IPv6 addresses out for free).

@ProbablePrime Feel free to reach out if you'd like any assistance.

**Comment by ProbablePrime** [comment]
I'm starting this today(Nice smooth onramp to the year i guess :P ). 

Working on getting all the LNL machines addressed and connectable by IPV6. Basically this, https://learn.microsoft.com/en-us/azure/load-balancer/ipv6-add-to-existing-vnet-powershell but modified for our use.

**Comment by ProbablePrime** [comment]
Ok, The JP Network node now has IPV6. That's great, in theory! But to get Resonite to connect to it we're gonna have to advertise the IPV6 Addresses on https://api.resonite.com/networknodes.

However, that presents an issue because it would mean multiple addresses are needed. 

Instead, i'm gonna see about swapping network nodes over to use a DNS'able name for their address. With that we can add AAAA entries for the network nodes. This will then be picked up by LNL(https://github.com/Yellow-Dog-Man/LiteNetLib/blob/d1b6feb6853e1fbc69329e09475d477bd4e564b4/LiteNetLib/NetUtils.cs#L40C21-L40C43)

**Comment by ProbablePrime** [comment]
https://www.reddit.com/r/ipv6/comments/1iox294/the_majority_of_traffic_in_the_united_states_to

**Comment by Stellanora64** [comment]
This has also been causing issues (textures not loading) for some Brazilian users.

**Comment by Stellanora64** [comment]
Textures not loading and staying checkerboard until they used a VPN.

Only a VPN fixed it after trying everything else in the checkerboard Textures guide on the wiki.

**Comment by ProbablePrime** [comment]
Thanks, that does help. Getting a log file would be good too though.

Asset retrieval should not be affected by IPV6, all the asset endpoints support IPv6, its just LNL that does not.

**Comment by ProbablePrime** [comment]
Updating this one to be purely about LNL as the others have been edited or adjusted by other things and I want to ensure new reports get new issues and that we can collate in a few other things.
